src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -83,15 +83,28 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
 
     }
 
+    // Helper method to create term queries with variations for spaces
+    private BoolQueryBuilder createNameMatchQuery(String field, String name) {
+        return QueryBuilders.boolQuery()
+            .should(QueryBuilders.termQuery(field, name)) // Exact match
+            .should(QueryBuilders.termQuery(field, " " + name)) // Prepend space
+            .should(QueryBuilders.termQuery(field, name + " ")) // Append space
+            .should(QueryBuilders.termQuery(field, " " + name + " ")); // Prepend and append space
```

Comment:
Have you tried out wildcard query? I think that will allow us to search against a regex and handle multiple spaces at the start or end